glog-src/demangle.cc

```diff
@@ -167,7 +167,7 @@ static size_t StrLen(const char *str) {
 // Returns true if "str" has at least "n" characters remaining.
 static bool AtLeastNumCharsRemaining(const char *str, int n) {
   for (int i = 0; i < n; ++i) {
-    if (str == '\0') {
+    if (str[i] == '\0') {
```

Comment:
Looks like g++ was right :)

Reply:
glog throws many "undefined reference to `pthread_rwlock_init'" without -pthread, so it doesn't seem to be an issue with your code at all.